TesteCategoria.java

```diff
@@ -0,0 +1,46 @@
+package casaDoCodigo;
+
+import java.util.LinkedList;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TesteCategoria {
+	@Test
+	public void testeCategoria() {
```

Comment:
*Avaliamos de forma lógica o nível de complicação de cada trecho de código. Aqui não tem espaço para feeling. Usamos a teoria da carga cognitiva e medimos o nível de pontos de entendimentos necessário por trecho.*

Esse é um trecho de teste que a gente entender for, loop, reassign de variáveis etc. Tudo isso vai aumentando os pontos de entendimento e, consequentemente, aumentando a dificuldade de entendimento em si. No caso da execução de testes automatizados, você pode criar um método anotado com ```@Test``` para cada caso de uso em si.